torch/_refs/__init__.py

```diff
@@ -906,13 +906,14 @@ def add(
     if alpha is not None:
         dtype = a.dtype if isinstance(a, TensorLike) else b.dtype  # type: ignore[union-attr]
         python_type = utils.dtype_to_type(dtype)
-        if not utils.is_weakly_lesser_type(type(alpha), python_type):
-            msg = (
-                "alpha argument of type {0} cannot be safely cast to type {1}!".format(
-                    type(alpha), python_type
-                )
-            )
-            raise ValueError(msg)
+        # TODO: this check is too restrictive. torch.add(bool_tensor, bool_tensor, 0) should work.
```

Comment:
cc @ngimel. This check triggers on `torch.add(bool_tensor, bool_tensor, 0)`, which works fine on eager. My understanding is that refs should work in all cases that eager works - do you happen to know the purpose of this check?

Reply:
We need to modify these checks to account for PyTorch allowing integer numbers with boolean tensors. Typically a number cannot be a "higher" Python type than the Python type corresponding to the tensors' type. For example,

```
a = torch.randint(0, 5, (2, 2))
b = torch.randint(0, 5, (2, 2))
torch.add(a, b, alpha=.5)
: RuntimeError: For integral input tensors, argument alpha must not be a floating point number.
```

The exception to this is that boolean tensors will interoperate with integers, and I think the integer is always cast to a bool before the operation. The references just don't implement this particular oddity yet.

Reply:
thanks for the overview! I'll update this check to pass in the bool case.